### PR TITLE
Created "Use-of-Unencrypted-URI-Schemes.bcheck" File.

### PR DESCRIPTION
**Pull request by LabMC**
### BCheck Contributions
* [X] BCheck compiles and executes as expected
* [X] BCheck contains appropriate metadata (name, version, author, description and appropriate tags)
* [X] Only .bcheck files have been added or modified
* [X] BCheck is in the appropriate folder
* [X] PR contains single or limited number of BChecks (Multiple PRs are preferred)
* [X] BCheck attempts to minimize false positives

#### **Note #1**: Uploaded "Use-of-Unencrypted-URI-Schemes.bcheck" File.
  - This BCheck attempts to look for hard-coded uses of unencrypted URI schemes (http, ftp, ldap, smtp without STARTTLS).
  - To reduce FPs, I included checks for attributes within hard-coded URLs (such as ldap:// being fine if the discovered URL uses port 636).
  - I also included several scanned file types & hard-coded file types to ignore (the most being .svg file types & use of w3.org).
  - I included several known URLs to ignore, such as known JavaScript library/framework links (jQuery, momentjs, sizzlejs).
  - I have the list of all hard-coded indicators for FPs to ignore at the beginning of the file & will likely update it based on future testing.
  - FP prevention also includes checking for "Content-Type: image/".
  - As a unique FP test, I tried to add to the regex a way to ignore "commented-out" JavaScript code.
    - URLs prior to /* within 98 characters (98 to set a boundary that won't be too excessive) will not be detected unless */ is also detected within 98 characters.
    - This method isn't perfect, but works well enough; the unlikely event wherein the sequence /* */ /* is used will still result in commented code being detected.
  - I tested the file numerous times to ensure the Regex patterning was works as intended on my machine.

#### **Note #2**: Deleted "InsecureContentSecurityPolicy.bcheck" File.
  - I thought I deleted this file already in my last pull request.
  - For reference, this is the original file that is a copy of "Content-Security-Policy.bcheck" in /archived.

#### **Note #3**: Edited "Content-Security-Policy.bcheck" File.
  - Last minor edits to the "Content-Security-Policy.bcheck" (added tags & updated language to beta-v2).

### CODE REVIEW

**Comment by Hannah-PortSwigger** [comment]
Thank you for your submission! We're in the process of reviewing your BCheck, and should either merge this or provide you with some feedback soon.

If there's anything else we can help with in the meantime, please let us know.

File: other/Use-of-Unencrypted-URI-Schemes.bcheck [hunk 1]
```diff
@@ -0,0 +1,102 @@
+metadata:
+    language: v2-beta
+    name: "Use of Unencrypted URI Schemes"
+    description: "This BCheck looks for unencrypted URI schemes ('http', 'ftp', 'ldap', 'smtp') within HTTP responses' bodies."
+    author: "Kyle Gilligan"
+    tags: "http", "ftp", "ldap", "smtp"
```
Comment (PortSwiggerWiener) [inline other/Use-of-Unencrypted-URI-Schemes.bcheck:6]: You could add passive here as well.

File: other/Use-of-Unencrypted-URI-Schemes.bcheck [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+# Commented-Out Regex:
+    # Recorded list of whitelisted variables to reduce known false positives.
```
Comment (PortSwiggerWiener) [inline other/Use-of-Unencrypted-URI-Schemes.bcheck:9]: Do we really need these comments?

Comment (PortSwiggerWiener) [reply other/Use-of-Unencrypted-URI-Schemes.bcheck:9]: Many thanks for your submission! It looks good. I've added a couple of really minor comments.

**Review by Hannah-PortSwigger — APPROVED** [review]
Looks good 👍

**Review by PortSwiggerWiener — APPROVED** [review]
Thanks for the revisions. 

Looks good. 👍

**Pull request merged**